### PR TITLE
feat(azure): document firmware CI

### PR DESCRIPTION
**Pull request by coderbyheart**

### CODE REVIEW

File: docs/aws/FirmwareContinuousIntegration.rst [hunk 1]
```diff
@@ -17,11 +17,11 @@ The CI run results in the following actions:
...
 1. Creating a new device and credentials on AWS IoT.
 #. Building a firmware that has the device ID hardcoded for the MQTT client ID.
-#. Scheduling a run on a self-hosted GitHub Actions runner. See :ref:`firmware_ci_runner_setup`.
+#. Scheduling a run on a self-hosted GitHub Actions runner. See :ref:`aws-firmware_ci_runner_setup`.
 #. Observing the firmware CI run until completion.
 #. Running assertions against the log result.
 
-The Firmware CI runner is running on a Raspberry Pi connected to GitHub Actions, where it receives the jobs to execute.
+The Firmware CI runner is running on a PC connected to GitHub Actions, where it receives the jobs to execute.
 Following are the actions performed by the Firmware CI runner:
```
Comment (peknis) [inline docs/aws/FirmwareContinuousIntegration.rst]:
```suggestion
The Firmware CI runner performs the following actions:
```

File: docs/azure/FirmwareContinuousIntegration.rst [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+The CI run results in the following actions:
+
+1. Creating a new device and credentials on AWS IoT.
+#. Building a firmware that has the device ID hardcoded for the MQTT client ID.
+#. Scheduling a run on a self-hosted GitHub Actions runner. See :ref:`azure-firmware_ci_runner_setup`.
+#. Observing the firmware CI run until completion.
+#. Running assertions against the log result.
+
+The Firmware CI runner is running on a PC connected to GitHub Actions, where it receives the jobs to execute.
+Following are the actions performed by the Firmware CI runner:
```
Comment (peknis) [inline docs/azure/FirmwareContinuousIntegration.rst]:
```suggestion
The Firmware CI runner performs the following actions:
```

File: docs/azure/FirmwareContinuousIntegration.rst [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+1. Programming the firmware and the optional credentials using the connected debugger to the connected nRF9160 DK or Thingy:91.
+#. Logging all output until one of the following conditions occur:
+
+   a. A timeout is reached
+   #. A stop condition is reached. (Wait for a log output to match a string.)
```
Comment (peknis) [inline docs/azure/FirmwareContinuousIntegration.rst]:
```suggestion
   * A timeout is reached
   * A stop condition is reached. (Wait for a log output to match a string.)
```
Do the same in the AWS doc.

File: docs/azure/FirmwareContinuousIntegration.rst [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+.. note::
+
+   These devices connect to the existing instance of the nRF Asset Tracker, so the firmware tests will not set up a new empty nRF Asset Tracker Azure environment for every test.
+   It runs against the production environment.
```
Comment (peknis) [inline docs/azure/FirmwareContinuousIntegration.rst]:
```suggestion
   The test runs against the production environment.
```
Do the same in the AWS doc.

File: docs/azure/FirmwareContinuousIntegration.rst [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+.. code-block:: bash
+
+   az ad sp create-for-rbac --name 'https://nrfassettracker.invalid/firmware-ci' \                         
+      --role contributor \
+      --scopes \
+         "/subscriptions/${SUBSCRIPTION_ID}/resourceGroups/${RESOURCE_GROUP:-nrfassettracker}" \
+      --sdk-auth \
+      > ci-credentials.json
+
+Assign access permissions to IoT hub for the role:
```
Comment (peknis) [inline docs/azure/FirmwareContinuousIntegration.rst]:
```suggestion
Assign access permissions to the IoT hub for the role: 
```

File: docs/azure/FirmwareContinuousIntegration.rst [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+      --assignee $(az ad sp list --display-name "https://nrfassettracker.invalid/firmware-ci" | jq -r '.[0].objectId') \
+      --role $(az role definition list --output json | jq -r '.[] | select(.roleName=="IoT Hub Data Contributor") | .id') \
+      --scope "/subscriptions/${SUBSCRIPTION_ID}/resourceGroups/${RESOURCE_GROUP}/providers/Microsoft.Devices/IotHubs/${APP_NAME}IotHub"
+
+.. _azure-firmware_ci_runner_setup:
+
+Firmware CI runner setup
+************************
+
+To set up Firmware CI runner, complete the following steps:
```
Comment (peknis) [inline docs/azure/FirmwareContinuousIntegration.rst]:
```suggestion
To set up the Firmware CI runner, complete the following steps:
```

File: docs/azure/FirmwareContinuousIntegration.rst [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+
+1. Prepare an environment that fulfills the :ref:`necessary system requirements <system-requirements>`.
+
+1. Download `JLink <https://www.segger.com/downloads/jlink/>`_ for your platform, and ensure that :file:`JLinkExe` is in your path.
```
Comment (peknis) [inline docs/azure/FirmwareContinuousIntegration.rst]:
```suggestion
#. Download `JLink <https://www.segger.com/downloads/jlink/>`_ for your platform, and ensure that :file:`JLinkExe` is in your path.
```
You don't need empty lines between the numbered items.

File: docs/azure/FirmwareContinuousIntegration.rst [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+
+#. Follow `the instruction about self-hosted runners <https://docs.github.com/en/actions/hosting-your-own-runners/about-self-hosted-runners>`_ to set up a runner for your firmware repository.
+   Ensure to use ``firmware-ci`` when prompted for the labels.
```
Comment (peknis) [inline docs/azure/FirmwareContinuousIntegration.rst]:
```suggestion
   Make sure to use ``firmware-ci`` when prompted for the labels.
```
Do the same in the AWS doc.

**Review by peknis — CHANGES_REQUESTED** [review]

File: docs/azure/FirmwareContinuousIntegration.rst [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+
+.. _azure-firmware_ci_runner_setup:
+
+Firmware CI runner setup
+************************
+
+To set up the Firmware CI runner, complete the following steps:
+
+1. Prepare an environment that fulfills the :ref:`necessary system requirements <system-requirements>`.
+
```
Comment (peknis) [inline docs/azure/FirmwareContinuousIntegration.rst]: You could still leave out the empty lines between the numbered items, but approving.

**Review by peknis — APPROVED** [review]

**Review by wiba-nordic — APPROVED** [review]

**Pull request merged**